File: GXpert/GXpert.Web/Modules/ServerTypes/LiveSessions/LiveSessionLogColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LiveSessionLogRow } from "./LiveSessionLogRow";

export interface LiveSessionLogColumns {
    Id: Column<LiveSessionLogRow>;
    Date: Column<LiveSessionLogRow>;
    StartTime: Column<LiveSessionLogRow>;
    EndTime: Column<LiveSessionLogRow>;
    TeacherPrn: Column<LiveSessionLogRow>;
    ClassTitle: Column<LiveSessionLogRow>;
    SubjectTitle: Column<LiveSessionLogRow>;
    InsertDate: Column<LiveSessionLogRow>;
    InsertUserId: Column<LiveSessionLogRow>;
    UpdateDate: Column<LiveSessionLogRow>;
    UpdateUserId: Column<LiveSessionLogRow>;
    IsActive: Column<LiveSessionLogRow>;
}

export class LiveSessionLogColumns extends ColumnsBase<LiveSessionLogRow> {
    static readonly columnsKey = 'LiveSessions.LiveSessionLog';
    static readonly Fields = fieldsProxy<LiveSessionLogColumns>();
}

File: GXpert/GXpert.Web/Modules/LiveSessions/LiveSessionLog/LiveSessionLogColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.LiveSessions.Columns;

[ColumnsScript("LiveSessions.LiveSessionLog")]
[BasedOnRow(typeof(LiveSessionLogRow), CheckNames = true)]
public class LiveSessionLogColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public DateTime Date { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public string TeacherPrn { get; set; }
    public string ClassTitle { get; set; }
    public string SubjectTitle { get; set; }
}